assessor/bin/assess-minibroker-charts.rb

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env ruby
+##
+# The purpose of this script is to assess which of the publicly
+# available charts for the databases supported by minibroker will work
+# with SCF.
+
+# Configuration
+# - Fixed location: `stable` helm repository
+# - Configurable:   work directory for state
+# - Configurable:   scf source directory
+# - Configurable:   SCF namespace
+# - Configurable:   Cluster Admin Password
+# - Configurable:   Operation mode (full, incremental)
+
+require 'fileutils'
+require 'net/http'
+require 'open3'
+require 'optparse'
+require 'uri'
+require 'yaml'
+
+require_relative "../lib/colors.rb"
+require_relative "../lib/config.rb"
+require_relative "../lib/helmrepo.rb"
+require_relative "../lib/locations.rb"
+require_relative "../lib/log.rb"
+require_relative "../lib/spawn.rb"
+require_relative "../lib/state.rb"
+require_relative "../lib/terminal.rb"
+require_relative "../lib/testcore.rb"
+
+@top = File.dirname(File.dirname(File.dirname(File.absolute_path(__FILE__))))
+
+def main
+  config
+  base_statistics
+
+  @assessed = 0
+  @skipped = 0
+  state
+
+  engines.each do |engine|
+    master_index[engine].each do |chart|
+      enginev       = chart['appVersion']
+      chartv        = chart['version']
+      chartlocation = chart['urls'].first
+
+      # We are ignoring all the entries for which we do not have the
+      # engine version. Because that is the plan id later, therefore
+      # required.
+      next unless enginev
+      next if skip_chart(engine, enginev, chartv)
+
+      assess_chart(chart, engine, enginev, chartv, chartlocation)
+    end
+  end
+
+  rewind
+  puts "#{"Skipped".cyan}:  #{@skipped}"  if @skipped
+  puts "#{"Assessed".cyan}: #{@assessed}" if @assessed
+end
+
+def master_index
+  unless @master
+    puts "#{"Retrieving".cyan} master index ..."
+    @master = helm_index(stable)['entries']
+  end
+  @master
+end
+
+def helm_index(location)
+    uri = URI.parse (location + "/index.yaml")
+    res = Net::HTTP.get_response uri
+
+    # Debugging, save index data.
+    File.write(File.join(@workdir, 'index-location.txt'), uri)
+    File.write(File.join(@workdir, 'index.yaml'), res.body)
+
+    YAML.load (res.body)
+end
+
+def base_statistics
+  engines.each do |engine|
+    # Debugging. Save per-engine indices.
+    File.write(File.join(@workdir, "e-#{engine}.yaml"), master_index[engine].to_yaml)
+
+    # We are ignoring all the entries for which we do not have the
+    # engine version. Because that is the plan id later, therefore
+    # required.
+
+    count = master_index[engine].select {|c| c['appVersion']}.length
+    puts "#{"Extracting".cyan} engine #{engine.blue}: #{count.to_s.cyan}"
```

Comment:
```suggestion
    puts "#{"Extracting".cyan} engine #{engine.blue}: #{count.to_s.cyan} versions"
```
Just makes it easier to read :)

Reply:
Actually `charts`.

Reply:
Fixed with commit dc68b80856848585810d27f16be4ba547b0cd313.